DependencyInjection/Factory/Resolver/WebPathResolverFactory.php

```diff
@@ -13,24 +13,22 @@
 
 use Symfony\Component\Config\Definition\Builder\ArrayNodeDefinition;
 use Symfony\Component\DependencyInjection\ContainerBuilder;
-use Symfony\Component\DependencyInjection\DefinitionDecorator;
 
-class WebPathResolverFactory implements ResolverFactoryInterface
+class WebPathResolverFactory extends AbstractResolverFactory
 {
     /**
      * {@inheritdoc}
      */
     public function create(ContainerBuilder $container, $resolverName, array $config)
     {
-        $resolverDefinition = new DefinitionDecorator('liip_imagine.cache.resolver.prototype.web_path');
+        $resolverDefinition = $this->getChildResolverDefinition();
         $resolverDefinition->replaceArgument(2, $config['web_root']);
         $resolverDefinition->replaceArgument(3, $config['cache_prefix']);
-        $resolverDefinition->addTag('liip_imagine.cache.resolver', array(
+        $resolverDefinition->addTag('liip_imagine.cache.resolver', [
             'resolver' => $resolverName,
-        ));
-        $resolverId = 'liip_imagine.cache.resolver.'.$resolverName;
+        ]);
 
-        $container->setDefinition($resolverId, $resolverDefinition);
+        $container->setDefinition($resolverId = 'liip_imagine.cache.resolver.'.$resolverName, $resolverDefinition);
```

Comment:
We do this often in the codebase now and can be found all over the place. While I disagree it makes it harder to read, I recognize many people feel this way. We should likely adopt a `CONTRIBUTING.md` file that defines some coding standards that the community agrees on as a whole.